draft-js-plugins-editor/src/Editor/index.js

```diff
@@ -66,23 +65,28 @@ class PluginEditor extends Component {
     let newEditorState = editorState;
     this.resolvePlugins().forEach((plugin) => {
       if (plugin.onChange) {
-        newEditorState = plugin.onChange(newEditorState);
+        newEditorState = plugin.onChange(newEditorState, this.getPluginMethods());
       }
     });
 
     if (this.props.onChange) {
-      this.props.onChange(newEditorState);
+      this.props.onChange(newEditorState, this.getPluginMethods());
     }
   };
 
   getEditorState = () => this.props.editorState;
+  getPluginMethods = () => ({
+    getPlugins: () => this.props.plugins.slice(0),
+    getProps: () => ({ ...this.props }),
+    setEditorState: this.onChange,
+    getEditorState: this.getEditorState,
+    getReadOnly: () => this.props.readOnly,
```

Comment:
Further down we have `readOnly={this.props.readOnly || this.state.readOnly}`. Why is `readOnly` here just from the props? Why would plugins use this instead of just taking it from `getProps`?